File: Maximum69Number.js
/**
 * @param {number} num
 * @return {number}
 */
var maximum69Number = function (num) {
  //convert the num into a string , and iterate through each char
  //if the element is equal to 9 go to the next character in sequence else flip and return as a number
  let str = num.toString().split("");

  for (let c = 0; c < str.length; c++) {
    if (str[c] === "6") {
      str[c] = "9";
      return Number(str.join(""));
    } else {
      continue;
    }
  }
  return str.join("");
};

File: maximum69Number.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        #convert the number to a string 
        #split the string into an array of elements
        #iterate through the string and look for the first instance of a 6 and then change it to a 9
        #convert the array back into a string and then into a number 
        #return
        num = str(num)
        num = list(num)
        for i in range(len(num)):
            if num[i] == '6':
                num[i] = '9'
                break
        num = "".join(num)
        return int(num)